pkg/client/apiutil/apimachinery.go

```diff
@@ -58,9 +88,36 @@ func RESTClientForGVK(gvk schema.GroupVersionKind, baseConfig *rest.Config, code
 	} else {
 		cfg.APIPath = "/apis"
 	}
-	cfg.NegotiatedSerializer = serializer.DirectCodecFactory{CodecFactory: codecs}
 	if cfg.UserAgent == "" {
 		cfg.UserAgent = rest.DefaultKubernetesUserAgent()
 	}
-	return rest.RESTClientFor(cfg)
+	return cfg
+
+}
+
+//Copied from deprecated-dynamic/bad_debt.go
+// dynamicCodec is a codec that wraps the standard unstructured codec
+// with special handling for Status objects.
+// Deprecated only used by test code and its wrong
+type dynamicCodec struct{}
+
+func (dynamicCodec) Decode(data []byte, gvk *schema.GroupVersionKind, obj runtime.Object) (runtime.Object, *schema.GroupVersionKind, error) {
+	obj, gvk, err := unstructured.UnstructuredJSONScheme.Decode(data, gvk, obj)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if _, ok := obj.(*metav1.Status); !ok && strings.ToLower(gvk.Kind) == "status" {
```

Comment:
interesting we have `Status` resource as well ?

Reply:
https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#Status

I believe this is used for errors from the API servers but I don't know for a fact.

Reply:
correct, the `Status` Kind is used to convey API errors.  It doesn't belong to any particular resource, IIRC.